plutus-core/plutus-ir/test/transform/knownCon/applicative

```diff
@@ -0,0 +1,128 @@
+  )
+  (termbind
+    (strict)
+    (vardecl ds1 [ Maybe (fun (con integer) (con integer)) ])
+    [
+      [
+        [
+          (let
+            (nonrec)
+            (typebind (tyvardecl b (type)) (fun (con integer) (con integer)))
+            (lam
+              dFunctor
+              [
+                (lam
+                  f
+                  (fun (type) (type))
+                  (all
+                    a
+                    (type)
+                    (all b (type) (fun (fun a b) (fun [ f a ] [ f b ])))
+                  )
+                )
+                Maybe
+              ]
+              (lam
+                f
+                (fun (con integer) b)
+                (lam
+                  fa
+                  [ Maybe (con integer) ]
+                  [ [ { { dFunctor (con integer) } b } f ] fa ]
+                )
+              )
+            )
+          )
+          (abs
+            a
+            (type)
+            (abs
+              b
+              (type)
+              (lam
+                f
+                (fun a b)
+                (lam
+                  ds2
+                  [ Maybe a ]
+                  {
+                    [
+                      [
+                        {
+                          [ { Maybe_match a } ds2 ]
+                          (all dead (type) [ Maybe b ])
+                        }
+                        (lam a a (abs dead (type) [ { Just b } [ f a ] ]))
+                      ]
+                      (abs dead (type) { Nothing b })
+                    ]
+                    (all dead (type) dead)
+                  }
+                )
+              )
+            )
+          )
+        ]
+        (lam
+          p
+          (con integer)
+          (lam q (con integer) [ [ (builtin addInteger) p ] q ])
+        )
+      ]
+      x
+    ]
+  )
+  {
+    [
+      [
+        {
+          [ { Maybe_match (fun (con integer) (con integer)) } ds1 ]
+          (all dead (type) [ Maybe (con integer) ])
+        }
+        (lam
+          g
+          (fun (con integer) (con integer))
+          (abs
+            dead
+            (type)
+            {
+              [
+                [
+                  {
+                    [ { Maybe_match (con integer) } [ { Just (con integer) } (con integer 2) ] ]
+                    (all dead (type) [ Maybe (con integer) ])
+                  }
+                  (lam
+                    v
+                    (con integer)
+                    (abs dead (type) [ { Just (con integer) } [ g v ] ])
+                  )
+                ]
+                (abs dead (type) { Nothing (con integer) })
+              ]
+              (all dead (type) dead)
+            }
+          )
+        )
+      ]
+      (abs dead (type) { Nothing (con integer) })
+    ]
+    (all dead (type) dead)
+  }
+)
+-- Adapted from `Budget.Spec.applicative`. The last `Maybe_match` scrutinizes
+-- a constructor application.
```

Comment:
at the top please

Do we need this test given that we have the `applicative` test elsewhere?

Reply:
This is not the applicative budget test. That test has no know constructors. This is modified so that one of the scrutinee is a known constructor.

Reply:
I guess my question is: what is this adding to the test suite?

Reply:
A more complex test case. Other test cases are nothing but destructor applications. This one has `Lam`, `TermBind` etc.

Reply:
Fair enough.